### PR TITLE
Iron deficiency risk effects

### PR DESCRIPTION
**Pull request by alibow**

### CODE REVIEW

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 1]
```diff
@@ -35,3 +35,198 @@ Iron Deficiency Risk Effects
...
+
+Instead, the relative risks should be applied to both the YLD (or incidence) and YLL (CSMR or EMR) rate for maternal disorders (or a specific modeled individual maternal disorder), such that:
+
+if :math:`Hgb_i >= Hgb_\text{GBD}:`
+
+.. math:: 
+
+    rate_i = rate_\text{GBD} * 1 / e^{\text{ln(RR)} * (Hgb_i - Hgb_\text{GBD})}
+
+if :math:`Hgb_i < Hgb_\text{GBD}:`
```
Comment (aflaxman) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:143]: If I understand correctly, these two branches are mathematically equivalent, because e^-x = 1/e^x

**Review by aflaxman — APPROVED** [review]

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 1 above)
Comment (alibow) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:143]: Great!! That was lost on me and I should have double checked. Thanks for pointing this out and I will update it.

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 2]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+
+.. todo::
+
+  Cite the CHERG iron report, GBD 2019 methods appendix
+
+.. note::
+
+  According to a correspondence with Chris Murray, there is contraversy surrounding the inclusion of this risk-outcome pair in GBD because there is evidence supporting an association between iron supplementation and hemoglobin increases as well as evidence supporting an association between hemoglobin increases and maternal mortality, but no/little evidence to support an association between iron supplementation and maternal mortality directly.
+
+Notably, the *same* relative risks for this risk-outcome pair apply to both YLDs and YLLs and are constant for each age-group and each most-detailed cause within the maternal disorders cause heirarchy; assuming that changes in hemoglobin concentration affect each of the maternal disorders equally and also affect morbidity and mortality equally.
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:79]: This sentence is a bit confusing to me, as the clause after the semicolon is not a complete sentence.  Is this second part just restating what the part before the semicolon means?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 3]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+
+  A limitation of this approach is that if changes in hemoglobin concentration affect a maternal disorder with a higher disability weight more or less than a maternal disorder with a lower disability weight, then our estimation of change in maternal disorder YLDs due to a change in hemoglobin concentration will be biased. 
+
+.. todo::
+
+  Include fiugre for cause hierarychy of maternal disorders
+
+The relative risks used for this risk outcome pair are scaled to units of hemoglobin concentration in g/dL. Because the iron deficiency exposure (and mean population hemoglobin concentration) in GBD 2019 are defined in g/L, the units will need to be converted.
+
+The PAF for this risk outcome pair is computed using the :code-block:`cont_paf_nocap` function found `here <https://stash.ihme.washington.edu/projects/RF/repos/paf/browse/math.R#9-10,30>`_, with the following parameter values:
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:89]: I think `:code-block:` is only a directive, not a role, so the `cont_paf_nocap` function is not rendering correctly.   I think `:code:` might be a role, or you can just use double backticks, i.e. \``cont_paf_nocap``

Also, it looks like the `math.R` code has been updated since this PR was created, so the specific lines in the link seem to be pointing to the wrong place. In fact, I don't even see a `cont_paf_nocap` function, just a `cont_paf` function...

Would it make more sense to find the older version of the file and point to that, or update to point to the newer version (which could be subject to additional changes that could break the link again...)?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 4]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+  
+  lower = 0
+  upper = 5_000
+  rr_scalar = 10
+  dist = 'normal'
+  inv_exp = 1
+
+.. note::
+
+  According to a correspondence with Nick Kassebaum, a normal distribution is assumed for the PAF calculation in this risk outcome pair, although a potential methodological improvement for GBD 2020 will be to assume the ensemble distribution used in the hemoglobin model instead.
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:101]: Do you mean that a normal distribution is assumed for the hemoglobin concentration in the population, even though they model it as an ensemble, or is something else assumed to be normally distributed?

Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:101]: If yes to the hemoglobin distribution, do you know why they use a normal instead for the PAF calculation?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 5]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+Maternal Disorders
+++++++++++++++++++++
+
+The recommended approach for modeling the relationship between hemoglobin and maternal disorders is described in the `Alternative Approach`_ section in this document. Therefore, there is no need to use the PAF in order to apply the relative risks.
+
+Instead, the relative risks should be applied to both the YLD (or incidence) and YLL (CSMR or EMR) rate for maternal disorders (or a specific modeled individual maternal disorder), such that:
+
+.. math:: 
+
+    rate_i = rate_\text{GBD} * 1 / e^{\text{ln(RR)} * (Hgb_i - Hgb_\text{GBD})}
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:139]: Where does this equation come from? Is this the standard way to use RR's for continuous risk exposures?  What is the definition of the RR, and what are its units?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 6]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+
+  Cite the CHERG iron report, GBD 2019 methods appendix
+
+.. note::
+
+  According to a correspondence with Chris Murray, there is contraversy surrounding the inclusion of this risk-outcome pair in GBD because there is evidence supporting an association between iron supplementation and hemoglobin increases as well as evidence supporting an association between hemoglobin increases and maternal mortality, but no/little evidence to support an association between iron supplementation and maternal mortality directly.
+
+Notably, the *same* relative risks for this risk-outcome pair apply to both YLDs and YLLs and are constant for each age-group and each most-detailed cause within the maternal disorders cause heirarchy; assuming that changes in hemoglobin concentration affect each of the maternal disorders equally and also affect morbidity and mortality equally.
+
+  A limitation of this approach is that if changes in hemoglobin concentration affect a maternal disorder with a higher disability weight more or less than a maternal disorder with a lower disability weight, then our estimation of change in maternal disorder YLDs due to a change in hemoglobin concentration will be biased.
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:81]: Is the extra indentation for this paragraph intentional, or should it be at the same level as the preceding paragraph?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 5 above)
Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:139]: Just based on dimensional analysis, it seems like maybe there's a constant missing or something?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 7]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+
+.. list-table:: Relative Risks
+   :widths: 5 5 5
+   :header-rows: 1
+
+   * - Exposure
+     - Relative Risk
+     - Note
+   * - Per unit deficit of hemoglobin (g/dL)
+     - 1.252472 (0.920269, 0.701621)
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:174]: Is this "value (lower, upper)", or something else?  Because the values in parentheses are both smaller than the initial number...

Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:174]: Also, I would like to see the definition of the relative risk written down somewhere so I can interpret what these numbers mean.

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 5 above)
Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:139]: @aflaxman Can you shed any light on this equation for me?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 6 above)
Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:81]: Also, this sentence is a bit long and hard to parse... not sure if there'd be a good way to simplify it.

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 5 above)
Comment (NathanielBlairStahn) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:139]: Also, I would recommend some slight formatting changes to get the LaTeX typesetter to interpret each word-variable as a unit rather than treating each letter as a separate variable, which is the default in math mode.  See if you like how any of these look (the only difference between them is how you want to write the exponential):

```
.. math::

    \mathrm{rate}_i = \mathrm{rate}_\text{GBD} \cdot 1 / e^{\ln(\mathit{RR}) \cdot (\text{Hgb}_i - \text{Hgb}_\text{GBD})}

.. math::

    \mathrm{rate}_i = \mathrm{rate}_\text{GBD} \cdot e^{-\ln(\mathit{RR}) \cdot (\text{Hgb}_i - \text{Hgb}_\text{GBD})}

.. math::

    \mathrm{rate}_i = \mathrm{rate}_\text{GBD} \cdot e^{\ln(\mathit{RR}) \cdot (\text{Hgb}_\text{GBD} - \text{Hgb}_i)}
```

If you change the formatting in the equation, the formatting of the variable names in the table should also be changed accordingly.

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 8]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+- Simulant maternal disorder state
+
+Simulants with higher hemoglobin concentrations should have lower maternal disorder rates (the specific difference in maternal disorder rates can be quanified for simulants with known hemoglobin concentrations).
+
+If intervention causes increase in hemoglobin, the intervention scenario rate of maternal disorders should be lower than the baseline rate of maternal disorders.
+
+Assumptions and Limitations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Notably, anemia is a sequela of the maternal disorder, maternal hemorrhage. Therefore, if a simulation model of an intervention that affects hemoglobin concentration evaluates both the impact on burden due to maternal disorders and the impact of YLDs due to anemia, it is possible that the impact on YLDs for anemia due to maternal hemorrhage will be double counted. This is likely a relative small portion of DALYs, but should be investigated and considered prior to implementation.
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:193]: How about saying "anemia is a sequela of maternal hemorrhage, which is a subcause of maternal disorders" for the first sentence?

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst [hunk 9]
```diff
@@ -35,3 +35,190 @@ Iron Deficiency Risk Effects
...
+.. list-table:: Parameter Definitions
+   :widths: 5 5 5
+   :header-rows: 1
+
+   * - Parameter
+     - Definition
+     - Note
+   * - :math:`Hgb_i`
+     - Hemoglobin value for an individual pregnant simulant in g/dL
+     - Needs unit conversion! (Exposure typically defined in g/L)
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:152]: Should we just go ahead and be more explicit about this? E.g. "Hgb value pulled from GBD are in g/L, so divide them by 10."

**Review by NathanielBlairStahn — APPROVED** [review]
Provisionally approving so we can get this merged -- just make a note to address the above issues in a future PR

File: docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst (hunk 4 above)
Comment (alibow) [reply docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/risk_effects/index.rst:101]: Yes, that is correct. I am not entirely sure, but it seemed like it was just a convenience thing that they weren't happy with and wanted to fix.
